src/daft-logical-plan/src/ops/project.rs

```diff
@@ -86,6 +86,25 @@ impl Project {
         input: Arc<LogicalPlan>,
         projection: Vec<ExprRef>,
     ) -> logical_plan::Result<(Arc<LogicalPlan>, Vec<ExprRef>)> {
+        // Check if projection contains any window functions, if so, return without factoring.
+        // This is because window functions may implicitly reference columns via the window spec.
+        let mut has_window = false;
+        projection.iter().any(|expr| {
+            expr.apply(|e| {
+                if matches!(e.as_ref(), Expr::Over(..)) {
+                    has_window = true;
+                    Ok(TreeNodeRecursion::Jump)
+                } else {
+                    Ok(TreeNodeRecursion::Continue)
+                }
+            })
+            .is_ok()
+        });
```

Comment:
The is_ok() check doesn't make sense. We want to check if has_window in the iter.any()
```suggestion
        projection.iter().any(|expr| {
            let mut has_window = false;
            expr.apply(|e| {
                if matches!(e.as_ref(), Expr::Over(..)) {
                    has_window = true;
                    Ok(TreeNodeRecursion::Jump)
                } else {
                    Ok(TreeNodeRecursion::Continue)
                }
            }).unwrap();
            
            has_window
        });
```

Reply:
Then you can just do `let has_window = // above`

Reply:
The advantage of this, other than code clarity, is that the iterator can early terminate